src/shogun/base/SGObject.h

```diff
@@ -124,6 +124,58 @@ class CSGObject
 	/** destructor */
 	virtual ~CSGObject();
 
+	/** serializes the SGObject to a binary file
+	 *
+	 * @param filename Binary archive filename
+	 */
+	void save_binary(const char* filename) const;
+
+	/** serializes the SGObject to a JSON file
+	 *
+	 * @param filename JSON archive filename
+	 */
+	void save_json(const char* filename) const;
+
+	/** serializes the SGObject to a XML file
+	 *
+	 * @param filename XML archive filename
+	 */
+	void save_xml(const char* filename) const;
+
+	/** loads SGObject from a Binary file
+	 *
+	 * @param filename Binary archive filename
+	 */
+	void load_binary(const char* filename);
+
+	/** loads SGObject from a JSON file
+	 *
+	 * @param filename JSON archive filename
+	 */
+	void load_json(const char* filename);
+
+	/** loads SGObject from a XML file
+	 *
+	 * @param filename XML archive filename
+	 */
+	void load_xml(const char* filename);
+
+#ifndef SWIG // SWIG should skip this part
+	/** serializes SGObject parameters to Archive with Cereal
+	 *
+	 * @param ar Archive
+	 */
+	template<class Archive>
```

Comment:
this works atm just because you include cereal header in `shogun/lib/common.h` which should be removed.... so you need to do a forward declaration of `Archive` there instead of the include to make this work.